### ISSUE TITLE
Redirect not working

### ISSUE THREAD

**Issue by facundofederico** [lunarway/openbanking#80]
I'm testing my prod integration and when creating a payment I get redirected to your authorization page. I still haven't tested making the payment, but if I decline or just wait it out, the page is not redirecting (and apparently not even detecting the decline event).

My callback address is localhost, so I think that could be a problem (although that address is registered)

**Comment by facundofederico** [comment]
I now also tried it in my staging environment and the problem persists. Your auth does not seem to be detecting those events correctly

**Comment by dentych** [comment]
Do you still require assistance with this?

**Comment by facundofederico** [comment]
Yes, it's still happening, even in production. If the user declines the payment, your auth page does not update and we have to wait for it to expire to be detected as cancelled

**Comment by AndersSoee** [comment]
@facundofederico we have just made some improvements to the redirect logic.
Does that resolve your issue?

Otherwise we'd need a information, so we can trace a specific login attempt

**Comment by facundofederico** [comment]
@AndersSoee I tried again and it still fails. 

Once I get to the https://auth.openbanking.prod.lunar.app/aisp-pisp/polling page, if I approve the payment everything goes well, but if I reject it the counter just goes on until it expires.

It sounds as if you were detecting the approval event but not the rejection one.

My payment query for this test was [poolingUrl]?paymentID=40beb3b3-f6ed-4c9a-9086-94abf3dbe29d&initiateTime=2025-06-06T12:25:31Z

**Comment by dentych** [comment]
This should now be working, please try again.
